microphone/drv_audio_pdm.c

```diff
@@ -101,8 +104,23 @@ static void drv_audio_pdm_event_handler(nrfx_pdm_evt_t const * const p_evt)
 			{
 				pdm_buf[l].released = true;
 				data_source_info.audio_source_info.audio_buffer = &pdm_buf[l].mic_buf;
+				if (l == 0 && timestamp_buffer[l] == 0) {
+					// First buffer, set timestamp to recording start timestamp
+					timestamp_buffer[l] = recording_start_timestamp;
+				} else {
+					// Subsequent buffers
+					// t_rec_i = t_release_i - (PDM_BUF_SIZE / effective_sampling_rate) * 1000
+					const uint32_t base_sample_rate = 20000; // 20kHz
+					uint64_t current_time = systick_get_millis();
+					uint32_t buffer_duration = (PDM_BUF_SIZE / base_sample_rate) * 1000;
```

Comment:
sample rate can be a compile-time constant, and the result from the operation `PDM_BUF_SIZE / base_sample_rate) * 1000` could be too due to const evaluation if this calc was in a macro instead. 

Not functionally different but I would expect that to yield better performance by a slight margin. 
Perhaps the calculation of these can be combined with this calculation: https://docs.nordicsemi.com/bundle/ps_nrf5340/page/pdm.html 
you could move the value for `pdfm_cfg.clock_freq` in `drv_audio_pdm.c` to a define. 

You could also take advantage that this now supports C23 and constexpr can be used